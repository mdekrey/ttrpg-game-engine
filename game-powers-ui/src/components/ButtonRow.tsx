import classNames from 'classnames';

export function ButtonRow({
	className,
	reversed = true,
	...props
}: JSX.IntrinsicElements['div'] & { reversed?: boolean }) {
	return (
		<div className={classNames(className, { 'flex-row-reverse': reversed }, 'flex justify-start gap-2')} {...props} />
	);
}
