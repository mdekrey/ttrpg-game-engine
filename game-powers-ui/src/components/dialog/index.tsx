export { useDialog, DialogProvider } from './context';
export { DisplayDialogs } from './DisplayDialogs';
