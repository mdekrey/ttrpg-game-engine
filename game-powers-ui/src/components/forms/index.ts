export * from './label/label';
export * from './select/select';
export * from './textbox/textbox';
export * from './ValidationMessages';
