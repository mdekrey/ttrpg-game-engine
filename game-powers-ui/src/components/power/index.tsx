export { FlavorText } from './FlavorText';
export { InfoBlock } from './InfoBlock';
export { MagicItemTable } from './MagicItemTable';
export { RulesText } from './RulesText';
export { Power } from './Power';
export { PowerTextBlock } from './PowerTextBlock';
export type { PowerTextBlockProps } from './PowerTextBlock';
