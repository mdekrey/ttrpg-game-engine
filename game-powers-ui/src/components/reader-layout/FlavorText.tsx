import classNames from 'classnames';

export function FlavorText({ children, className, ...props }: JSX.IntrinsicElements['p']) {
	return (
		<p className={classNames(className, 'font-flavor font-bold italic')} {...props}>
			{children}
		</p>
	);
}
