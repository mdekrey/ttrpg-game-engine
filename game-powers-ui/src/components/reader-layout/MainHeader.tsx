import classNames from 'classnames';

export function MainHeader({ children, className, ...props }: JSX.IntrinsicElements['h1']) {
	return (
		<h1 className={classNames(className, 'font-header font-bold mt-4 first:mt-0 text-theme text-3xl')} {...props}>
			{children}
		</h1>
	);
}
