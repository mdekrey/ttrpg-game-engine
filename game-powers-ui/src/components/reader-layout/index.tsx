import classNames from 'classnames';
import { MdxComponents } from 'components/layout/mdx-components';
import { ReactNode } from 'react';

export const ReaderLayout = ({ children, className }: { children?: ReactNode; className?: string }) => (
	<div className={classNames('my-4 mx-4 md:mx-auto print:col-count-2 max-w-2xl print:max-w-none font-text', className)}>
		<MdxComponents>{children}</MdxComponents>
	</div>
);
