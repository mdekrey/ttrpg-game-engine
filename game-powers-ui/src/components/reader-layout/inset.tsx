import { BlockquoteHTMLAttributes, DetailedHTMLProps } from 'react';
import classNames from 'classnames';
import { recurse } from 'core/jsx/recurse';
import { pipeJsx } from 'core/jsx/pipeJsx';
import { infoFontTemplate } from 'components/layout/mdx-components';

export function Inset({
	children,
	className,
	...props
}: DetailedHTMLProps<BlockquoteHTMLAttributes<HTMLQuoteElement>, HTMLQuoteElement>) {
	return (
		<blockquote
			className={classNames(className, 'bg-gradient-to-r from-tan-fading p-2 my-4')}
			style={{ pageBreakInside: 'avoid' }}
			{...props}>
			{pipeJsx(<>{children}</>, recurse(infoFontTemplate))}
		</blockquote>
	);
}
