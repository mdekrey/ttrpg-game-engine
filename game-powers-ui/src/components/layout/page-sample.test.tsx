import { render } from '@testing-library/react';
import Page from 'components/layout/page-sample.mdx';
import { ReaderLayout } from 'components/reader-layout';

test('renders', () => {
	const { container } = render(
		<ReaderLayout>
			<Page />
		</ReaderLayout>
	);
	expect(container).toMatchSnapshot();
});
