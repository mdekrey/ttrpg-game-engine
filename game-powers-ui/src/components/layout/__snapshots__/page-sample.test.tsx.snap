// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
<div
class="my-4 mx-4 md:mx-auto print:columns-2 max-w-2xl print:max-w-none font-text"
>
<page-sample.mdx />
</div>
</div>
`;
